edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -197,19 +208,69 @@ const CoursePage: ApplicationRoute = () => {
 
 	const renderProfessorModifButton = useCallback(() => {
 		return (
-			<TTouchableOpacity style={{
-				position: 'absolute',
-				bottom: 20,
-				right: 20,
-				zIndex: 1000,
-			}} onPress={handleButtonPress}>
+			<TTouchableOpacity
+				testID={testIDs.addElementsTouchable}
+				style={{
+					position: 'absolute',
+					bottom: 20,
+					right: 20,
+					zIndex: 1000,
+				}}
+				onPress={handleButtonPress}
+			>
 				<Icon name="add-circle" size={60} color='blue' />
 			</TTouchableOpacity>
 		);
 	}, []);
+
+	const addAssignmentCallback = useCallback((assignment: Assignment) => {
+		addAssignmentAction(id as CourseID, JSON.stringify(assignment));
+		setModalVisible(false);
+	}, [id]);
+
+	const addMaterialCallback = useCallback((material: Material) => {
+		addMaterialAction(id as CourseID, JSON.stringify(material));
+		setModalVisible(false);
+	}, [id]);
+
+	const updateAssignmentCallback = useCallback((assignment: Assignment, assignmentID?: AssignmentID) => {
+		if (assignmentID) {
+			updateAssignmentAction(id as CourseID, assignmentID, JSON.stringify(assignment));
+		}
+		setAssignmentToEdit(null);
+		setModalEditAssignmentVisible(false);
+	}, [id]);
+
+	const updateMaterialCallback = useCallback((material: Material, materialID?: MaterialID) => {
+		if (materialID) {
+			updateMaterialAction(id as CourseID, materialID, JSON.stringify(material));
+		}
+		setMaterialToEdit(null);
+		setModalEditMaterialVisible(false);
+	}, [id]);
+
+	const updateCourseCallback = useCallback((course: UpdateCourseArgs) => {
+		updateCourseAction(id as CourseID, JSON.stringify(course));
+		setModalParamVisible(false);
+	}, [id]);
+
+	const removeAssignmentCallback = useCallback((assignmentID: AssignmentID) => {
+		removeAssignmentAction(id as CourseID, assignmentID);
+		setAssignmentToEdit(null);
+		setModalEditAssignmentVisible(false);
+	}, [id]);
+
+	const removeMaterialCallback = useCallback((materialID: MaterialID) => {
+		removeMaterialAction(id as CourseID, materialID);
+		setMaterialToEdit(null);
+		setModalEditMaterialVisible(false);
+	}, [id]);
```

Comment:
That's an interresting approach you have here.
I see that you provide kind of duplicate actions.
I don't know if it's possible to do it in typescript, but I would have a suggestion as an enhancement for the future, is to create an `AbstractCourseContentEditor` where you put all the common ground function and create 2 instances for the `Assignment` and `Material`, this way you reduce significantly boilerplate code and leverage the DRY principle.